From f736b93e8a6a68faf234640c61ced280b353611c Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Mon, 8 Sep 2014 10:05:31 -0700
Subject: [PATCH] Fix assumption about hwmon0

The logic looks for coretemp in all possible paths including old
kernel sysfs path for core temp and /sys/class/hwmon. If fails
it assumes hwmon0, which is not correct as a fallback. This
may contain some device which has no relation to cpu temp.
---
 src/thd_engine_default.cpp | 40 +++++++---------------------------------
 1 file changed, 7 insertions(+), 33 deletions(-)

diff --git a/src/thd_engine_default.cpp b/src/thd_engine_default.cpp
index 4b39b8e..5fab7c8 100644
--- a/src/thd_engine_default.cpp
+++ b/src/thd_engine_default.cpp
@@ -183,21 +183,7 @@ int cthd_engine_default::read_thermal_sensors() {
 	}
 	if (index == sensor_count) {
 		// No coretemp sysfs exist, try hwmon
-		thd_log_warn("Thermal DTS: No coretemp sysfs, trying hwmon \n");
-		cthd_sensor *sensor = new cthd_sensor(index,
-				"/sys/class/hwmon/hwmon0/temp1_input", "hwmon",
-				SENSOR_TYPE_RAW);
-		if (sensor->sensor_update() != THD_SUCCESS) {
-			delete sensor;
-			return THD_ERROR;
-		}
-		sensors.push_back(sensor);
-		++index;
-
-		if (index == sensor_count) {
-			thd_log_error("Thermal DTS or hwmon: No Zones present: \n");
-			return THD_FATAL_ERROR;
-		}
+		thd_log_warn("Thermal DTS: No coretemp sysfs found!!\n");
 	}
 
 	// Add from XML sensor config
@@ -297,24 +283,7 @@ int cthd_engine_default::read_thermal_zones() {
 		}
 
 		if (!cpu_zone_created) {
-			// No coretemp sysfs exist, try hwmon
-			thd_log_warn("Thermal DTS: No coretemp sysfs, trying hwmon \n");
-
-			cthd_zone_cpu *zone = new cthd_zone_cpu(count,
-					"/sys/class/hwmon/hwmon0/", 0);
-			if (zone->zone_update() == THD_SUCCESS) {
-				zone->set_zone_active();
-				zones.push_back(zone);
-				++count;
-				cpu_zone_created = true;
-			} else {
-				delete zone;
-			}
-
-			if (!cpu_zone_created) {
-				thd_log_error("Thermal DTS or hwmon: No Zones present: \n");
-				return THD_FATAL_ERROR;
-			}
+			thd_log_error("Thermal DTS or hwmon: No Zones present Need to configure manually\n");
 		}
 	}
 	zone_count = count;
@@ -450,6 +419,11 @@ int cthd_engine_default::read_thermal_zones() {
 #endif
 	zone_count = count;
 
+	if (!zone_count) {
+		thd_log_info("No Thermal Zones found \n");
+		return THD_FATAL_ERROR;
+	}
+
 	def_binding.do_default_binding(cdevs);
 
 	for (unsigned int i = 0; i < zones.size(); ++i) {
-- 
1.9.3

